rfcs/0027-formalize-documentation-organization.md

```diff
@@ -0,0 +1,430 @@
+    * -> Work With microTVM
+    * ->  microTVM Architecture
+  * VTA (index to existing docs)
+* Developer Guide
+  * TVM Contributor Tutorial
+    * …
+  * How To
+    * Write an operator
+    * Write a backend
+    * …
+* TVM Architecture Guide
+  * Architecture Overview
+  * Research Papers
+  * Frontend
+  * Relay: Graph-level design: IR, pass, lowering
+  * TensorIR: Operator-level design: IR, schedule, pass, lowering
+  * TOPI: Pre-defined operators operator coverage
+  * AutoScheduler / AutoTVM: Performance tuning design
+  * Runtime & microTVM design
+  * Customization with vendor libraries BYOC workflow
+  * RPC system
+  * Target system
+* API Reference (reference)
+  * Generated C++ Docs…
+  * Generated Python Docs…
+* Index
+
+## Document Code Organization
+
+This refactor will require a shift of how the documents are organized. In
+general, Tutorials and How-Tos are written as Sphinx Gallery documents,
+allowing for the generation of text, python source, and Jupyter Notebooks. This
+allows the user to consume these working code samples in a number of ways, but
+comes at the cost of fixed format that can be confusing to navigate. To help
+mitigate this, the tutorials and how-tos will be broken up into a more fine
+grained directory structure. For example:
+
+    tvm/
+      gallery/
+        dev_how_tos/
+          compile_models/
+          ...
+        how_tos/
+        tutorial/
+
+Rather than render the gallery in one pass as a nested structure (resulting in
+a single page with multiple sections), instead each directory will be rendered
+independently. This will aid in navigation through the galleries, and also give
+more fine-grained grouping of similar topics. The naming of the directory
+reflects the organization of Sphinx documentation folder, for example:
+
+    tvm/
+      docs/
+        deep_dive/
+        how_tos/
+          index.rst
+          **compile_models/**
+          ...
+        reference/
+        **tutorial/**
+        dev_deep_dive/
+        dev_how_tos/
+        dev_reference/
+        dev_tutorial/
+
+Depending on the type of documentation, some of the directories may be
+generated. For example, the tutorial and compile_models directories are
+auto-generated by Sphinx Gallery. To add a new Sphinx Gallery requires the
+following steps:
+
+1. Create a gallery subdirectory with the how-to or tutorial documents
+2. Create entries in the docs conf.py example_dirs and gallery_dirs variables
+   to reference the source and target directories.
+3. Update the appropriate index pages in the docs subdirectories to add the new
+   directories to the Sphinx table of contents.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+One consistent drawback of this approach is how major subprojects are handled.
+For example, microTVM may require a specific set of tutorials and how-tos, but
+these can become mixed in with other TVM specific documents. This will be
+mitigated through two means:
+
```

Comment:
Maybe remove this newline if "two means" refers to the first two sentences of the next paragraph?

Reply:
Fixed.